packages/mui-base/src/Slider/SliderThumb/SliderThumb.types.ts

```diff
@@ -0,0 +1,93 @@
+import { defaultRenderFunctions } from '../../utils/defaultRenderFunctions';
+import { BaseUIComponentProps } from '../../utils/types';
+import { SliderRootOwnerState, UseSliderReturnValue } from '../Root/SliderRoot.types';
+
+export interface SliderThumbOwnerState extends SliderRootOwnerState {}
+
+export interface SliderThumbProps
+  extends Partial<Omit<UseSliderThumbParameters, 'rootRef'>>,
+    Omit<BaseUIComponentProps<'span', SliderThumbOwnerState>, 'render'> {
+  onPointerLeave?: React.PointerEventHandler;
+  onPointerOver?: React.PointerEventHandler;
+  onBlur?: React.FocusEventHandler;
+  onFocus?: React.FocusEventHandler;
+  onKeyDown?: React.KeyboardEventHandler;
+  /**
+   * A function to customize rendering of the component.
+   */
+  render?:
+    | ((
+        props: React.ComponentPropsWithRef<'span'>,
+        inputProps: React.ComponentPropsWithRef<'input'>,
+        state: SliderThumbOwnerState,
+      ) => React.ReactElement)
+    | (React.ReactElement & { ref: React.Ref<Element> })
+    | keyof typeof defaultRenderFunctions;
+}
```

Comment:
`renderProp` shouldn't accept strings. In all the other components it's either a function or a React node.

Besides, can you think of any use cases that require the inner input to be customized? Other components, such as the Switch just render the inner input (as an internal implementation detail). Perhaps we should do the same here?

Reply:
> can you think of any use cases that require the inner input to be customized?

@michaldudak The special handling here is not for customizing the input, but for allowing a custom Thumb component to have children (e.g. putting the `Output` inside the thumb) while using a render function, to prevent custom children from overwriting the internal `input`. Unless you're doing both those things, the `render` prop works just like the other components!

I've updated the logic and docs on this a bit in https://github.com/mui/base-ui/pull/373/commits/70339a3866f84b2ef50727bdbce1b03c0d17fcc6

Reply:
It should be possible anyway with the standard `useComponentRenderer`, right?
`render={(props) => <CustomThumb {...props}><SomeOtherComponent />{props.children}</SomeOtherComponent></CustomThumb>}`.
I suppose it would just require some changes to the short form of the render prop (`render={<CustomThumb><SomeOtherComponent /></CustomThumb}`), to merge the children with internal children.

Reply:
> I suppose it would just require some changes to the short form of the render prop `(render={<CustomThumb><SomeOtherComponent /></CustomThumb})`, to merge the children with internal children.

I am already doing this here ~ https://github.com/mui/base-ui/blob/fbe5ef11e2b7a703325ace695232712cd2254c6f/packages/mui-base/src/Slider/Thumb/SliderThumb.tsx#L99-L110

> It should be possible anyway with the standard useComponentRenderer, right?
`render={(props) => <CustomThumb {...props}><SomeOtherComponent />{props.children}</SomeOtherComponent></CustomThumb>}`

I'm imagining the usage to be more like this:
```jsx
function MyThumb(props: any) {
  return (
    <Slider.Thumb
      {...props}
      render={(renderProps) => (
        <Wrapper {...renderProps}>
          {props.children}
          {renderProps.children}
        </Wrapper>
      )}
    />
  );
}

...

<MyThumb><MyCustomChild /></MyThumb>
```
rather than:
```jsx
<Slider.Thumb
  render={(renderProps) => (
    <MyThumb {...renderProps}>
      <CustomChild />
      {renderProps.children}
    </MyThumb>
  )}
/>
```
(though both ways should work)

And on second thought looking at the snippet above, I reverted the last change as I think explicitly having `inputProps` in the render signature is clearer (although a bit longer) than trying to hide the input in `renderProps.children`:
```jsx
<Slider.Thumb
  render={(props, _ownerState, inputProps) => {
    return (
      <MyCustomThumb {...props}>
        {props.children}
        <input {...inputProps}>
      <MyCustomThumb/>
    )
  }}>
/>
```

And in the Thumb component, this:
`return render(thumbProps, inputProps, ownerState)`
feels cleaner than this:
`return render({ ...thumbProps, children: <input {...inputProps} /> }, ownerState)`